presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/viewmodels/MediaViewModel.kt

```diff
@@ -0,0 +1,74 @@
+package com.safetyheads.akademiaandroida.presentation.ui.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.safetyheads.akademiaandroida.domain.entities.Media
+import com.safetyheads.akademiaandroida.domain.entities.Settings
+import com.safetyheads.akademiaandroida.domain.repositories.SettingsRepository
+import com.safetyheads.akademiaandroida.domain.usecases.GetAllVideoUseCase
+import com.safetyheads.akademiaandroida.domain.usecases.GetInstagramImageUseCase
+import kotlinx.coroutines.launch
+
+class MediaViewModel(
+    private val getInstagramImageUseCase: GetInstagramImageUseCase,
+    private val getAllVideoUseCase: GetAllVideoUseCase,
+    private val settingsRepository: SettingsRepository
+) : ViewModel() {
+
+    init {
+        getAllVideosFromYouTube()
+        getInstagramImage()
+    }
+
+    val youtubeFilmList: MutableLiveData<List<Media>> = MutableLiveData()
+    val instagramImageList: MutableLiveData<List<Media>> = MutableLiveData()
+    val connectedList: MutableLiveData<List<Media>> = MutableLiveData()
+
+    fun getInstagramImage() {
+        viewModelScope.launch {
+            getInstagramImageUseCase.invoke().collect() { imageList ->
+                val tempImageResult = imageList.getOrNull()
+                if (tempImageResult != null) {
+                    instagramImageList.value = tempImageResult.orEmpty()
+                    instagramImageList.postValue(tempImageResult.orEmpty())
+                    getConnectedList()
+                }
+            }
+        }
+    }
+
+    fun getAllVideosFromYouTube() {
+        viewModelScope.launch {
+            getAllVideoUseCase.invoke().collect() { videoList ->
+                val tempVideoResult = videoList.getOrNull()
+                if (tempVideoResult != null) {
+                    youtubeFilmList.value = tempVideoResult.orEmpty()
+                    youtubeFilmList.postValue(tempVideoResult.orEmpty())
+                    getConnectedList()
+                }
+            }
+        }
+    }
+
+    fun getConnectedList() {
```

Comment:
This function doesn't return anything so `get` isn't not best word here to describe what this function do.